course7/homeworks/fred/complex-rpn/main.cc

```diff
@@ -0,0 +1,65 @@
+#include "complex.h"
+#include "calculator.h"
+#include "utils.h"
+
+#include <iostream>
+#include <stdexcept>
+
+void insert(RpnCalculator& calculator, Complex complex) {
+  calculator.insert(complex);
+}
+
+void insert(RpnCalculator& calculator, Operator op) {
+  auto result = calculator.insert(op);
+  std::cout << "Result so far: " << result << std::endl;
+}
+
+int main(int argc, char** args) {
+  std::cout << "Welcome!" << std::endl;
+  std::cout << "Type 'q' to quit" << std::endl << std::endl;
+
+  RpnCalculator calculator;
+
+  std::string input = "";
```

Comment:
There is no point initializing the value to ```""```. The default-constructed ```std::string``` is empty.
Have you read the documentation of std::string? If not, do it!